### PR TITLE
Lesson 16 Exercises

### PR DESCRIPTION
**Pull request by gildedgardenia**

### CODE REVIEW

File: src/chapters/orm/exercises.rst
Comment (jimflores5) [inline src/chapters/orm/exercises.rst]: VSCode flagged "No newline at end of file."

File: src/chapters/orm/exercises.rst
Comment (jimflores5) [inline src/chapters/orm/exercises.rst]: Remember to put a clever title after "Exercises".  Maybe "Exercises: ORM, OMG!"

File: src/chapters/orm/exercises.rst
Comment (jimflores5) [inline src/chapters/orm/exercises.rst]: To remain consistent with the previous model exercises, add a link to the GitHub repository here.

File: src/chapters/orm/exercises.rst
Comment (jimflores5) [inline src/chapters/orm/exercises.rst]: With the number of class names, field names, and view names listed on this page, I can see the students getting very confused when they read these steps. This is especially true since the SQL lessons come between these exercises and the last time the students worked on ``coding-events``.

I think it would be helpful to include a screenshot or some form of benchmark about how the app should behave once the changes are made.

File: src/chapters/orm/exercises.rst
Comment (jimflores5) [inline src/chapters/orm/exercises.rst]: Similar to the previous section, I think some sort of "sanity check" would work well here.  Thinking as a student, I'm not 100% sure about the desired outcome.  I see *what* I need to do, but not *why*.

File: src/chapters/orm/exercises.rst
Comment (jimflores5) [inline src/chapters/orm/exercises.rst]: Clarification for the "uses" term:
1) "...that uses the string..." 
or 
2) "...``title``, and assign it the value ``"Create Category"``."

File: src/chapters/orm/exercises.rst
Comment (jimflores5) [inline src/chapters/orm/exercises.rst]: See previous "uses" comment.

File: src/chapters/orm/exercises.rst
Comment (jimflores5) [inline src/chapters/orm/exercises.rst]: What does "appropriately" mean in this case?  How will a student know if they have or have not done this?

File: src/chapters/orm/exercises.rst
Comment (jimflores5) [inline src/chapters/orm/exercises.rst]: Definitely add screenshots for the appearance of the final product.

**Review by jimflores5 — CHANGES_REQUESTED** [review]
Sally,
    As always, you do good work.  I only found a couple of phrasing issues.

As I read through the exercises, however, I felt more of a studio vibe.  I think some more hand-holding would be good as the students work through the steps.  Not hints, per se, but more along the lines of "here is how your app should behave" after each section.

File: src/chapters/orm/index.rst
Comment (chrisbaylewis) [inline src/chapters/orm/index.rst]: For the TOC listing, it would be better to spell out the acronym: "Object-Relational Mapping"

Comment (chrisbaylewis) [reply src/chapters/orm/index.rst]: Also, we might want to talk about chapter labeling for this chapter and the next, since the next one will also contain ORM-related stuff. I'll add that to working meeting notes.

File: src/chapters/orm/exercises.rst
Comment (gildedgardenia) [reply src/chapters/orm/exercises.rst]: Done! I fixed that!

File: src/chapters/orm/exercises.rst
Comment (gildedgardenia) [reply src/chapters/orm/exercises.rst]: Done! Thanks!

File: src/chapters/orm/exercises.rst
Comment (gildedgardenia) [reply src/chapters/orm/exercises.rst]: Done! Thanks!

File: src/chapters/orm/exercises.rst
Comment (gildedgardenia) [reply src/chapters/orm/exercises.rst]: Done! Thanks!

File: src/chapters/orm/exercises.rst
Comment (gildedgardenia) [reply src/chapters/orm/exercises.rst]: Done! Thanks!

File: src/chapters/orm/exercises.rst
Comment (gildedgardenia) [reply src/chapters/orm/exercises.rst]: They have done this before. I added a link to the section where they learned how to use error validation for anyone who wants to review it.

File: src/chapters/orm/index.rst
Comment (gildedgardenia) [reply src/chapters/orm/index.rst]: Relabelled!

File: src/chapters/orm/exercises.rst
Comment (gildedgardenia) [reply src/chapters/orm/exercises.rst]: Added a note to the top with the relevant links for students for what these classes should look like. I want to link to the text sections, but since they are in a different branch, I linked straight to Youtube for now. I also added a section on what the final app will look like.

File: src/chapters/orm/exercises.rst
Comment (gildedgardenia) [reply src/chapters/orm/exercises.rst]: I commented above!

File: src/chapters/orm/exercises.rst
Comment (gildedgardenia) [reply src/chapters/orm/exercises.rst]: Done! Thanks!

File: src/chapters/orm-part1/exercises.rst
Comment (jimflores5) [inline src/chapters/orm-part1/exercises.rst]: Upon the second review, I am re-thinking my "add a GitHub link" comment.  With the Angular work, many students forked and cloned the repository multiple times instead of just checking out the specified branch.  This led to confusion and code mishaps.

To avoid this, it might be better to remind the students that they already have the repository, so they only need to checkout instead of download something new.  (Assuming they set up their fork to pull in changes from the our original).

File: src/chapters/orm-part1/exercises.rst
Comment (jimflores5) [inline src/chapters/orm-part1/exercises.rst]: Add a comma after "database".

File: src/chapters/orm-part1/exercises.rst
Comment (jimflores5) [inline src/chapters/orm-part1/exercises.rst]: Unneeded article.  "Create ``EventCategoryController`` in..."

File: src/chapters/orm-part1/exercises.rst
Comment (jimflores5) [inline src/chapters/orm-part1/exercises.rst]: How about shifting the internal link from "section" to "error validation"?

File: src/chapters/orm-part1/exercises.rst
Comment (jimflores5) [inline src/chapters/orm-part1/exercises.rst]: This sentence is a little awkward to read (passive, two -ing endings, two "already", and "stored storage").

File: src/chapters/orm-part1/exercises.rst
Comment (jimflores5) [inline src/chapters/orm-part1/exercises.rst]: "...we added..."

"...our app to test it."

File: src/chapters/orm-part1/exercises.rst
Comment (jimflores5) [inline src/chapters/orm-part1/exercises.rst]: Add a comma after the ``localhost`` address, and change "table as" to "table has".

Phrasing suggestion: "...your table now contains the newest event category!"

**Review by jimflores5 — APPROVED** [review]
Sally, the screenshots really clarify the goal of the exercises.  Nice work.

I found a few more comma and typo issues, and I re-thought my earlier GitHub link comment.  Once these are resolved, feel free to merge!

File: src/chapters/orm-part1/exercises.rst
Comment (gildedgardenia) [reply src/chapters/orm-part1/exercises.rst]: Done! Thanks!

File: src/chapters/orm-part1/exercises.rst
Comment (gildedgardenia) [reply src/chapters/orm-part1/exercises.rst]: Done! Thanks!

File: src/chapters/orm-part1/exercises.rst
Comment (gildedgardenia) [reply src/chapters/orm-part1/exercises.rst]: Done! Thanks!

File: src/chapters/orm-part1/exercises.rst
Comment (gildedgardenia) [reply src/chapters/orm-part1/exercises.rst]: Done! Thanks!

File: src/chapters/orm-part1/exercises.rst
Comment (gildedgardenia) [reply src/chapters/orm-part1/exercises.rst]: Done! Thanks!

File: src/chapters/orm-part1/exercises.rst
Comment (gildedgardenia) [reply src/chapters/orm-part1/exercises.rst]: Done! Thanks!

File: src/chapters/orm-part1/exercises.rst
Comment (gildedgardenia) [reply src/chapters/orm-part1/exercises.rst]: Changed the wording to let students know their code should resemble that branch to get started.

Comment (jimflores5) [reply src/chapters/orm-part1/exercises.rst]: The wording revision sounds good.

**Pull request merged**
